lib/addressProvider/SMLAddressProvider.js

```diff
@@ -0,0 +1,56 @@
+const DAPIAddress = require('./DAPIAddress');
+
+class SMLAddressProvider {
+  /**
+   *
+   * @param {SMLProvider} smlProvider
+   * @param {ListAddressProvider} listAddressProvider
+   */
+  constructor(smlProvider, listAddressProvider) {
+    this.smlProvider = smlProvider;
+    this.listAddressProvider = listAddressProvider;
+  }
+
+  /**
+   * Get random live DAPI address from SML
+   *
+   * @returns {Promise<DAPIAddress>}
+   */
+  async getLiveAddress() {
+    const sml = await this.smlProvider.getSimplifiedMNList();
+    const validMasternodeList = sml.getValidMasternodesList();
+
+    const allAddresses = this.listAddressProvider.getAllAddresses();
+
+    const updatedAddresses = validMasternodeList.map((smlEntry) => {
+      let address = allAddresses
+        .find((a) => smlEntry.proRegTxHash === a.getProRegTxHash());
```

Comment:
I think it's better to use hashmap here, as doing this will invoke find function 12 million times for 5k masternodes